### DISCUSSION TITLE
Pull Request and Merge

### DISCUSSION THREAD

**Discussion by akileus0000** [community/community#148108] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

Hi,

I'm gonna invite someone to our repo to make some changes on that.

However, I would like to make him only enable to create a new pull request on github, but not merge feature.

How should I do this?

**Comment by bitmanvs** [comment, marked as answer]
Hi @akileus0000,

Here're some steps to do that:

Go to Your Repository Settings
Navigate to your repository on GitHub.
Click on Settings.
Branch Protection Rules

In the left sidebar, go to Branches under Code and Automation.
Click on Add branch protection rule.
Set Up the Rule for Your Main Branch (e.g., main or master)

In the Branch name pattern, enter your default branch name (main or master).
Enable Required Settings

Check Require a pull request before merging.
Enable Require approvals (you can set it to at least 1 approval).
Disable Allow force pushes.
Disable Allow deletions.
Restrict Who Can Push to Matching Branches

Scroll to Restrict who can push to matching branches.
Add only the people (or team) you trust to merge pull requests.
Do not add the invited developer here.
Save Changes

Click on Save Changes.
Adjust the Developer’s Role (Optional)

Go to Settings → Collaborators and Teams.
Make sure the developer only has Write access, not Admin or Maintain.

**Reply by akileus0000** [reply]
Hi @bitmanvs 

I could get some feelings to do that through your comments.
